packages/core/src/connectivity/scp-cf/destination/destination-from-service.ts

```diff
@@ -220,21 +226,69 @@ class DestinationFromServiceRetriever {
     return credentials[0];
   }
 
-  private async addOAuthSamlAuth(
-    destination: Destination,
-    destinationOrigin: DestinationOrigin
+  private fetchDestinationByToken(
```

Comment:
If this function returns a promise, it should be marked as async
```suggestion
  private async fetchDestinationByToken(
```